staker/bold/bold_staker.go

```diff
@@ -83,43 +97,35 @@ type StateProviderConfig struct {
 var DefaultStateProviderConfig = StateProviderConfig{
 	ValidatorName:          "default-validator",
 	CheckBatchFinality:     true,
-	MachineLeavesCachePath: "/tmp/machine-leaves-cache",
+	MachineLeavesCachePath: filepath.Join(homeDir, conf.PersistentConfigDefault.GlobalConfig, "machine-hashes-cache"),
```

Comment:
Right now even if someone changes their GlobalConfig path it won't be reflected in this path, as it only uses the default. We should have the value for this config default to just "machine-hashes-cache", and if it's not configured to an absolute path join it to the GlobalConfig path.

Alternatively, if we pass through the "stack" (a geth *node.Node) we can call `stack.ResolvePath(config.MachineLeavesCachePath)` which will basically do the same thing as above.